simple_https_server/src/lib.rs

```diff
@@ -0,0 +1,180 @@
+#![allow(non_upper_case_globals)]
+
+// Import hacspec and all needed definitions.
+use std::{io::prelude::*, net::TcpListener, str, time::Duration};
+
+use base::{AppError, RecordStream};
+use bertie::{tls13api::*, tls13utils::*};
+#[cfg(feature = "evercrypt")]
+use evercrypt_cryptolib::*;
+#[cfg(not(feature = "evercrypt"))]
+use hacspec_cryptolib::*;
+use hacspec_lib::*;
+use rand::*;
+
+const SHA256_Aes128Gcm_EcdsaSecp256r1Sha256_X25519: Algorithms = Algorithms(
+    HashAlgorithm::SHA256,
+    AeadAlgorithm::Aes128Gcm,
+    SignatureScheme::EcdsaSecp256r1Sha256,
+    NamedGroup::X25519,
+    false,
+    false,
+);
+
+/*
+const SHA256_Chacha20Poly1305_RsaPssRsaSha256_X25519: Algorithms = Algorithms(
```

Comment:
These should all get into a list so we don't have commented code around. But we can do this in a followup.

Reply:
Yeah. I tried to do only a small amount of changes necessary to extract the functionality we need. Will cleanup later.

Reply:
https://github.com/cryspen/bertie/issues/50